sdk/state/ingest.go

```diff
@@ -108,3 +112,33 @@ func (c *Channel) ingestTxToUpdateUnauthorizedCloseAgreement(tx *txnbuild.Transa
 
 	return nil
 }
+
+// ingestTxMetaToUpdateBalances uses the transaction result meta data
+// from a transaction response to update local and remote escrow account
+// balances.
+func (c *Channel) ingestTxMetaToUpdateBalances(resultMetaXDR string) {
+	// If not a valid resultMetaXDR string, return.
+	var txMeta xdr.TransactionMeta
+	err := xdr.SafeUnmarshalBase64(resultMetaXDR, &txMeta)
+	if err != nil {
+		return
```

Comment:
We should return the error if we can't decode. That seems like a problem to me, rather than something we'd ignore.

Reply:
Respond to https://github.com/stellar/experimental-payment-channels/pull/234#issuecomment-902237782:

> 2\. I'm not 100% on `ingestTxMetaToUpdateBalances` not throwing an error if the xdr is bad. Follows the pattern of the other ingest functions, but if the accounts were updated on the network but the xdr string is messed up for whatever reason we'd probably want to signal that with an error to the `agent` or whoever calls this

I think in this case the error should be returned. We only ignore conditions that aren't errors, where it's information that is irrelevant. In this case we don't know if the information is irrelevant because we can't decode it.

Reply:
https://github.com/stellar/experimental-payment-channels/pull/234/commits/39ba4560aa27873a6fc6c14037dd5870d696039c